src/states/node.rs

```diff
@@ -2482,20 +2482,32 @@ impl Node {
                 debug!("{:?} Disconnecting from timed out peer {:?}", self, peer_id);
                 let _ = self.crust_service.disconnect(peer_id);
             }
-            let transition = self.purge_invalid_rt_entries(outbox);
+
+            let transition = if cfg!(feature = "use-mock-crust") {
+                Transition::Stay
+            } else {
+                self.purge_invalid_rt_entries(outbox)
```

Comment:
as now the ticking is enabled, this means purge will be called in the process of `connecting` (waiting for the un-acked msg to establish tunnel or else), and will panic directly.

purge will still be called (to carry out consistency check) once network is settled (no more msg exchange)